test/distributed/test_c10d_functional_native.py

```diff
@@ -720,6 +720,17 @@ def func(args: List[torch.Tensor]) -> torch.Tensor:
         out = AOTIRunnerUtil.run("cuda", func, (args,))
         torch.cuda.synchronize()
 
+    @unittest.skipIf(not HAS_GPU, "This is a GPU test!")
+    @fresh_inductor_cache()
+    def test_wait_tensor_temp(self):
+        def func(arg: torch.Tensor) -> torch.Tensor:
+            return funcol.wait_tensor(arg)
```

Comment:
I believe in the current unit test, `arg` is a normal torch.Tensor and calling `.wait_tensor()` on it has unclear semantics. Would it work if we pass an AsyncCollectiveTensor into this function (e.g. passing in the output of `funcol.all_reduce(buf0, "avg", "0")` from eager mode)?

Reply:
Discussed offline. I am landing this now to unblock https://github.com/pytorch/pytorch/pull/136534. @benjaminglass1  please follow up on the unit test improvement.

Reply:
@yf225 @desertfire See #137692